src/extension/utils.js

```diff
@@ -303,42 +312,62 @@ export async function assembleConfig({
   };
 }
 
-export async function addConfig(input, cb) {
-  const config = await assembleConfig(input);
-  const {
-    owner, repo, mountpoints,
-  } = config;
-  getState(({ configs }) => {
-    if (!configs.find((cfg) => owner === cfg.owner && repo === cfg.repo)) {
-      configs.push(config);
-      setConfig('sync', { hlxSidekickConfigs: configs })
-        .then(() => (typeof cb === 'function' ? cb(true) : null))
-        .then(() => log.info('added config', config))
-        .catch((e) => log.error('error adding config', e));
-      if (!mountpoints[0]) {
-        alert(i18n('config_add_no_mountpoint'));
-      } else {
-        alert(i18n('config_add_success'));
-      }
-    } else {
-      alert(i18n('config_project_exists'));
-      if (typeof cb === 'function') cb(false);
+export async function getProject(project) {
+  const { owner, repo } = project;
+  return getConfig('sync', `${owner}/${repo}`);
+}
+
+export async function setProject(project, cb) {
+  const { owner, repo } = project;
+  const handle = `${owner}/${repo}`;
+  const obj = {};
+  obj[handle] = project;
+  await setConfig('sync', obj, async () => {
+    // update project index
+    let projects = await getConfig('sync', 'hlxSidekickProjects');
+    if (!projects) projects = [];
+    if (!projects.includes(handle)) {
+      projects.push(handle);
+      await setConfig('sync', { hlxSidekickProjects: projects });
     }
   });
+  log.info('updated project', project);
+  if (typeof cb === 'function') {
+    cb(project);
+  }
+}
+
+export async function addProject(input, cb) {
+  const config = await assembleProject(input);
+  const project = await getProject(config);
+  if (!project) {
+    await setProject(config);
+    log.info('added project', config);
+    alert(i18n('config_add_success'));
+    if (typeof cb === 'function') cb(true);
+  } else {
+    log.info('project already exists', project);
+    alert(i18n('config_project_exists'));
+    if (typeof cb === 'function') cb(false);
+  }
 }
 
-export async function deleteConfig(i, cb) {
+export async function deleteProject(i, cb) {
```

Comment:
I find this a very fragile way of deleting by index. I think a `deleteProject(handle)` would be better.